Userland/truncate.cpp

```diff
@@ -0,0 +1,106 @@
+#include <LibCore/CArgsParser.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+enum TruncateOperation {
+    OP_Set,
+    OP_Grow,
+    OP_Shrink,
+};
+
+int main(int argc, char** argv)
+{
+    CArgsParser args_parser("truncate");
+
+    args_parser.add_arg("s", "size", "Resize the target file to (or by) this size. Prefix with + or - to expand or shrink the file, or a bare number to set the size exactly.");
+    args_parser.add_arg("r", "reference", "Resize the target file to match the size of this one.");
+    args_parser.add_required_single_value("file");
+
+    CArgsParserResult args = args_parser.parse(argc, argv);
+
+    if (!args.is_present("s") && !args.is_present("r")) {
+        args_parser.print_usage();
+        return -1;
+    }
+
+    if (args.is_present("s") && args.is_present("r")) {
+        args_parser.print_usage();
+        return -1;
+    }
+
+    auto op = OP_Set;
+    int size = 0;
+
+    if (args.is_present("s")) {
+        auto str = args.get("s");
+
+        switch (str[0]) {
+        case '+':
+            op = OP_Grow;
+            str = str.substring(1, str.length() - 1);
+            break;
+        case '-':
+            op = OP_Shrink;
+            str = str.substring(1, str.length() - 1);
+            break;
+        }
+
+        bool ok;
+        size = str.to_int(ok);
+        if (!ok) {
+            args_parser.print_usage();
+            return -1;
+        }
+    }
+
+    if (args.is_present("r")) {
+        struct stat st;
+        int rc = lstat(args.get("r").characters(), &st);
```

Comment:
Does it make sense to use `lstat` instead of `stat` here?
Shouldn't we follow symlinks instead of using the size of the symlink itself?

Reply:
Ooooh, yes. Yes we should. Very good catch.